test/base_test_fixture.h

```diff
@@ -871,30 +871,44 @@ struct base_test_fixture
 
         check_rows_equal(execute(connection, query), 10);
 
+        REQUIRE(connection.transactions() == 0);
         {
             nanodbc::transaction transaction(connection);
+            REQUIRE(connection.transactions() == 1);
             execute(connection, NANODBC_TEXT("delete from transaction_test;"));
             check_rows_equal(execute(connection, query), 0);
+            REQUIRE(connection.transactions() == 1);
             // ~transaction() calls rollback()
         }
+        REQUIRE(connection.transactions() == 0);
 
         check_rows_equal(execute(connection, query), 10);
 
+        REQUIRE(connection.transactions() == 0);
         {
             nanodbc::transaction transaction(connection);
+            REQUIRE(connection.transactions() == 1);
             execute(connection, NANODBC_TEXT("delete from transaction_test;"));
             check_rows_equal(execute(connection, query), 0);
-            transaction.rollback();
+            REQUIRE(connection.transactions() == 1);
+            transaction.rollback(); // only requests rollback performed in ~transaction()
+            REQUIRE(connection.transactions() == 1); // transaction not released yet
```

Comment:
Althouth the RAII makes sense, my intuition expected the symmestry with the `commit`. Actually, the `commit` doc is incorrect:

```
    //! \brief Marks this transaction for commit.
    void commit();
```
It should rather read "Commits transaction immediately".

I don't request any changes. If, however, we wanted to improve the current interface, I would suggest to use named rollback modes:

```
enum class RollbackPoint { ScopeExit, Immediate };
void rollback(RollbackPoint when = ScopeExit);
```

Boolean parameters, especially with default values, make code less readable.
Specifically, if I saw `rollback(false)` my :eyes: could have rolled :)